### PR TITLE
Bridge-Removal - miscellaneous cleanups

### PR DESCRIPTION
**Pull request by Yuqi-Du**
Background: one step further to the bridge-removal

# Dependencies related
- quarkus-smallrye-openapi, quarkus-smallrye-fault-tolerance (openapi related)
- quarkus-security (used by principle-header auth)
- quarkus-grpc (quarkus grpc support)
- quarkus-config-yaml (This is the one we use to resolve embedding-providers-config.yaml)
- add jakarta.validation dependency
- add quarkus-resteasy-reactive (this is for injecting jakarta.ws.rs.core.SecurityContext in DataAPIRequestInfo)
- and quarkus-resteasy-reactive-jackson (this is to remove the warning msg noise: `[WARNING] [io.quarkus.resteasy.reactive.server.deployment.QuarkusServerEndpointIndexer] Quarkus detected the use of JSON in JAX-RS method 'io.stargate.sgv2.jsonapi.api.v1.NamespaceResource#postCommand' but no JSON extension has been added. Consider adding 'quarkus-resteasy-reactive-jackson' (recommended) or 'quarkus-resteasy-reactive-jsonb'.`)
- caffeine (cache)
- jbool_expressions dependency (used when building cql expressions)
- add quarkus-hibernate-validator dependency (Enables validation of configuration, beans, REST endpoints, etc.)

# other cleanups
- add loggingConstants, add log-console format
- disable quarkus http basic auth,
- add quarkus-application ui-header
- add ReservedKeywords class, which is used by ColumnUtils
- add http access log pattern, "%h %l %t \"%r\" %s %b" 
- add openAPI configuration (checked generated openAPI files by fileDiff)
- add swaggerUi configuration (checked http://localhost:8183/swagger-ui/, also added stargate logo.png to META-INF)
- Delete CreateCollectionResolverVectorDisabledTest, DisableVectorSearchProfile, since they rely on stargate data-store config, and to check if vectorSearch is enabled. This will be deprecated after we remove quarkus-common module dependency, and this is just a redundant check because we only deploy data-api in vector kubes.

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,6 +74,27 @@
...
       <artifactId>quarkus-smallrye-health</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-openapi</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-fault-tolerance</artifactId>
+      <version>3.6.1</version>
```
Comment (tatu-at-datastax) [inline pom.xml:84]: Is this version needed? Does not come from managedDependency (quarkus-bom)?

Comment (Yuqi-Du) [reply pom.xml]: deleted

File: src/main/java/io/stargate/sgv2/jsonapi/config/constants/LoggingConstants.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.stargate.sgv2.jsonapi.config.constants;
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/config/constants/LoggingConstants.java:1]: @Yuqi-Du Why we need this?

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -110,10 +133,33 @@ quarkus:
   # adapt path of the open api definitions
   smallrye-openapi:
     path: /api/json/openapi
+    info-title: ${quarkus.application.name}
+    info-version: ${quarkus.application.version:}
+    info-description:
+    info-terms-of-service:
+    info-contact-name: Stargate
```
Comment (maheshrajamani) [inline src/main/resources/application.yaml:140]: Should this be not the stargate details instead say and point to data-api documentation

Comment (Yuqi-Du) [reply src/main/resources/application.yaml:140]: Certainly, I could improve these.
Similarly to other documentations, currently just focus on making sure no functionality mistake.

File: src/main/java/io/stargate/sgv2/jsonapi/config/constants/LoggingConstants.java (hunk 2 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/config/constants/LoggingConstants.java:1]: Yes, there is one constant used somewhere.
Others are redundant, these are just constant, can delete later

**Review by tatu-at-datastax — APPROVED** [review]

**Pull request merged**